core/src/net/cidr/v4.rs

```diff
@@ -53,20 +115,49 @@ impl Cidr for Ipv4Cidr {
         self.address
     }
 
+    #[inline]
+    fn contains(&self, address: Self::Addr) -> bool {
+        self.prefix == (self.mask & u32::from(address))
+    }
+
+    #[inline]
+    fn network(&self) -> Self::Addr {
+        Ipv4Addr::from(self.mask & u32::from(self.address))
+    }
+
+    #[inline]
+    fn broadcast(&self) -> Self::Addr {
+        Ipv4Addr::from((self.mask & u32::from(self.address)) | !self.mask)
+    }
+
     #[inline]
     fn length(&self) -> usize {
         self.length
     }
 
     #[inline]
-    fn new(address: Self::Addr, length: usize) -> Result<Self, CidrParseError> {
+    fn hostmask(&self) -> Self::Addr {
+        if let Some(mask) = u32::max_value().checked_shr(self.length as u32) {
```

Comment:
this would just be `IpAddr::from(!self.mask)`.

Reply:
derp. yep.